x-pack/plugins/actions/server/builtin_action_types/case/schema.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const MappingActionType = schema.oneOf([
+  schema.literal('nothing'),
+  schema.literal('overwrite'),
+  schema.literal('append'),
+]);
+
+export const MapRecordSchema = schema.object({
+  source: schema.string(),
+  target: schema.string(),
+  actionType: MappingActionType,
+});
+
+export const CaseConfigurationSchema = schema.object({
+  mapping: schema.arrayOf(MapRecordSchema),
+});
+
+export const ConnectorPublicConfiguration = {
```

Comment:
Instead of calling it connector, we can define an `ExternalIncidentManagementConfiguration`, so it will make more sense when we are moving inside of the case plugin.

Reply:
do we also need to have `ConnectorPublicConfiguration` and `ConnectorPublicConfigurationSchema` separated like that?

Reply:
`ConnectorPublicConfiguration` is a shared configuration between external services. Jira, for example, extends its own configuration on top of the common one.